src/common/mfu_util.c

```diff
@@ -1105,6 +1110,239 @@ bool mfu_is_lustre(const char* path)
+    char*            mfu_mi_mountpoint;
+    char*            mfu_mi_fstype;
+    char*            mfu_mi_fssubtype;
+    char*            mfu_mi_mountopts;
+} mfu_mount_info_t;
+
+void mfu_mount_info_destroy(struct mfu_mount_info* mip)
+{
+    if (mip == NULL) {
+        return;
+    }
+
+    mfu_free(&mip->mfu_mi_source);
+    mfu_free(&mip->mfu_mi_mountpoint);
+    mfu_free(&mip->mfu_mi_fstype);
+    mfu_free(&mip->mfu_mi_fssubtype);
+    mfu_free(&mip->mfu_mi_mountopts);
+}
+
+int mfu_mount_info_parse_fstype_line(
+    char*                  line,
+    size_t                 size,
+    struct mfu_mount_info* mip)
+{
+    char* saveptr = NULL;
+    char* found = strtok_r(line, ".", &saveptr);
+    if (found == NULL) {
+        return(ENOENT);
+    }
+
+    mip->mfu_mi_fstype = MFU_STRDUP(found);
+
+    found = strtok_r(NULL, ".", &saveptr);
+    if (found == NULL) {
+        return(0);
+    }
+
+    mip->mfu_mi_fssubtype = MFU_STRDUP(found);
+
+    return(0);
+}
+
+int
+mfu_mount_info_parse_mount_line(
+    char*                  line,
+    size_t                 size,
+    struct mfu_mount_info* mip)
+{
+    int num = 0;
+    char* saveptr = NULL;
+    char* found = strtok_r(line, " ", &saveptr);
+    while (found != NULL) {
+        num++;
+        switch (num) {
+            case 1:
+                mip->mfu_mi_source = MFU_STRDUP(found);
+                break;
+            case 2:
+                mip->mfu_mi_mountpoint = MFU_STRDUP(found);
+                break;
+            case 3:
+                mfu_mount_info_parse_fstype_line(found, strlen(found)+1, mip);
+                break;
+            case 4:
+                mip->mfu_mi_mountopts = MFU_STRDUP(found);
+                break;
+            default:
+                break;
+        }
+
+        found = strtok_r(NULL, " ", &saveptr);
+    }
+
+    return(0);
+}
+
+int mfu_mount_info_get_filesystem_list(struct list_head* mfu_mi_list)
+{
+    if (mfu_mi_list == NULL) {
+        return(EINVAL);
+    }
+
+    errno = 0;
+    FILE* mountfp = fopen("/proc/self/mounts", "r");
+    if (mountfp == NULL) {
+        return(errno);
+    }
+
+    char* line = NULL;
+    size_t len = 0;
+    ssize_t bytes = getline(&line, &len, mountfp);
+    while (bytes != -1)
+    {
+        struct mfu_mount_info* itemp = MFU_CALLOC(1, sizeof(struct mfu_mount_info));
+        if (itemp == NULL) {
+            fclose(mountfp);
+            return(ENOMEM);
+        }
+        INIT_LIST_HEAD(&itemp->mfu_mi_linkage);
+
+        /*
+         * Parse one of the lines in /proc/self/mounts
+         */
+        mfu_mount_info_parse_mount_line(line, len, itemp);
+
+        /*
+         *  Get the device for the mountpoint in question.
+         */
+        errno = 0;
+        struct stat st;
+        int rc = stat(itemp->mfu_mi_mountpoint, &st);
+        if (rc == -1) {
+            fclose(mountfp);
+            fprintf(stderr, "Could not stat: %s (%d)", itemp->mfu_mi_mountpoint, errno);
```

Comment:
I think you wanted the fprintf()’s changed to MFU_LOG(MFU_LOG_ERR, …)

Reply:
Good catch.  I'll update that.  And for errors, I also like to print the ``strerror(errno)``.